xarray/namedarray/core.py

```diff
@@ -415,7 +473,7 @@ def _nonzero(self) -> tuple[Self, ...]:
     def _as_sparse(
         self,
         sparse_format: str | Default = _default,
-        fill_value=dtypes.NA,
+        fill_value: typing.Any = dtypes.NA,
```

Comment:
The return type of this is wrong since the underlying duck array is now sparse.

Same with `_to_dense`